sui_programmability/examples/games_with_chance/sources/satoshi_flip.move

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+
+/// Satoshi flip is a fair and secure way to conduct a coin flip.
+/// A player called "house" chooses a random secret and uploads its hash on chain
+/// Another player called "player",  makes a guess on a predetermined bit (eg. last) of the secret
+/// House reveals the secret and the winner is determined
+/// 
```

Comment:
I'd a comment about the optimistic version, let's call the game (in comments) "Optimistic Satoshi Dice" and explain that the main difference between the original Satoshi dice algorithm used in Bitcoin is that if the house refuses to reveal the secret on time, player can win the game.